extensions/amp-fx-collection/0.1/test/integration/test-amp-fx-fly-in.js

```diff
@@ -0,0 +1,159 @@
+  `;
+
+  const extensions = ['amp-fx-collection'];
+
+
+  // Can't test a default `fly-in-left` animation as the
+  // `data-fly-in-distance` and `data-duration` parameters differ/device.
+  const bodyLeft = `
+    <div class="spacer"></div>
+    <div id="animTarget"
+      amp-fx="fly-in-left"
+      data-duration="500ms"
+      data-fly-in-distance="25%"
+      data-margin-start="20%">
+    </div>
+    <div class="spacer"></div>
+  `;
+
+  describes.integration("amp-fx='fly-in-left'", {
+    body: bodyLeft,
+    css,
+    extensions,
+  }, env => {
+
+    let win;
+    beforeEach(() => {
+      win = env.win;
+      toggleExperiment(win, 'amp-fx-fly-in', true, false);
+    });
+
+    it('runs fly-in-left animation with default parameters', () => {
+      expect(isExperimentOn(win, 'amp-fx-fly-in')).to.be.true;
+      const initialLeft = getComputedLeft(win);
+      win.scrollTo(0, 0.5 * getViewportHeight(win));
+      return win.setTimeout(() => {
+        expect(getComputedLeft(win)).to.be.above(initialLeft);
+      }, 2000);
+    });
+  });
+
+  // Can't test a default `fly-in-right` animation as the
+  // `data-fly-in-distance` and `data-duration` parameters differ/device.
+  const bodyRight = `
+    <div class="spacer"></div>
+    <div id="animTarget"
+      amp-fx="fly-in-right"
+      data-duration="500ms"
+      data-fly-in-distance="25%"
+      data-margin-start="20%">
+    </div>
+    <div class="spacer"></div>
+  `;
+
+  describes.integration("amp-fx='fly-in-right'", {
+    body: bodyRight,
+    css,
+    extensions,
+  }, env => {
+
+    let win;
+    beforeEach(() => {
+      win = env.win;
+      toggleExperiment(win, 'amp-fx-fly-in', true, false);
+    });
+
+    it('runs fly-in-right animation with default parameters', () => {
+      expect(isExperimentOn(win, 'amp-fx-fly-in')).to.be.true;
+      const initialLeft = getComputedLeft(win);
+      win.scrollTo(0, 0.5 * getViewportHeight(win));
+      return win.setTimeout(() => {
+        expect(getComputedLeft(win)).to.be.below(initialLeft);
+      }, 2000);
+    });
+  });
+
+  // Can't test a default `fly-in-bottom` animation as the
+  // `data-fly-in-distance` and `data-duration` parameters differ/device.
+  const bodyBottom = `
+    <div class="spacer"></div>
+    <div id="animTarget"
+      amp-fx="fly-in-bottom"
+      data-duration="500ms"
+      data-fly-in-distance="25%"
+      data-margin-start="20%">
+    </div>
+    <div class="spacer"></div>
+  `;
+
+  describes.integration("amp-fx='fly-in-bottom'", {
+    body: bodyBottom,
+    css,
+    extensions,
+  }, env => {
+
+    let win;
+    beforeEach(() => {
+      win = env.win;
+      toggleExperiment(win, 'amp-fx-fly-in', true, false);
+    });
+
+    it('runs fly-in-bottom animation with default parameters', () => {
+      expect(isExperimentOn(win, 'amp-fx-fly-in')).to.be.true;
+      const initialTop = getComputedTop(win);
+      win.scrollTo(0, 0.5 * getViewportHeight(win));
+      return win.setTimeout(() => {
```

Comment:
Ah you still need to return a Promise so the test knows when it's done. You could keep your original `timeout` utility function, but at the call site you'd call 
```js
return timeout(2000).then(() => { /* ... */ });
```
instead of 
```js
return Promise.resolve().then(timeout(2000)).then(() => { /* ... */ });
```

Reply:
Ooohh good catch - there is no wait for the timeout to finish. 

Changed the code to retain the original code with your edits. Thanks @cvializ!